package/MDAnalysis/analysis/base.py

```diff
@@ -220,7 +224,176 @@ def __setstate__(self, state):
+        :class:`MDAnalysis.analysis.base.AnalysisBase`.
+
+    Example
+    -------
+    .. code-block:: python
+
+        import MDAnalysis as mda
+        from MDAnalysis.analysis.rdf import InterRDF
+        from MDAnalysis.analysis.base import AnalysisCollection
+        from MDAnalysisTests.datafiles import TPR, XTC
+
+        u = mda.Universe(TPR, XTC)
+
+        # Select atoms
+        ag_O = u.select_atoms("name O")
+        ag_H = u.select_atoms("name H")
+
+        # Create individual analysis instances
+        rdf_OO = InterRDF(ag_O, ag_O)
+        rdf_OH = InterRDF(ag_H, ag_H)
+
+        # Create a collection for common trajectory
+        collection = AnalysisCollection(rdf_OO, rdf_OH)
+
+        # Run the collected analysis
+        collection.run(start=0, stop=100, step=10)
+
+        # Results are stored in the individual instances
+        print(rdf_OO.results)
+        print(rdf_OH.results)
+
+
+    .. versionadded:: 2.5.0
+
+    """
+
+    def __init__(self, *analysis_instances):
+        for analysis_object in analysis_instances:
+            if analysis_instances[0]._trajectory != analysis_object._trajectory:
+                raise ValueError(
+                    "`analysis_instances` do not have the same trajectory."
+                )
+            if not isinstance(analysis_object, AnalysisBase):
+                raise AttributeError(
+                    f"Analysis object {analysis_object} is "
+                    "not a child of `AnalysisBase`."
+                )
+
+        self._analysis_instances = analysis_instances
+
+    def run(
+        self,
+        start=None,
+        stop=None,
+        step=None,
+        frames=None,
+        verbose=None,
+        reset_timestep=True,
+    ):
+        """Perform the calculation
+
+        Parameters
+        ----------
+        start : int, optional
+            start frame of analysis
+        stop : int, optional
+            stop frame of analysis
+        step : int, optional
+            number of frames to skip between each analysed frame
+        frames : array_like, optional
+            array of integers or booleans to slice trajectory; `frames` can
+            only be used *instead* of `start`, `stop`, and `step`. Setting
+            *both* `frames` and at least one of `start`, `stop`, `step` to a
+            non-default value will raise a :exc:`ValueError`.
+        verbose : bool, optional
+            Turn on verbosity
+        reset_timestep : bool, optional
+            Reset the timestep object after for each ``analysis_object``.
+            Setting this to ``False`` can be useful if an ``analysis_object``
+            is performing a trajectory manipulation which is also useful for the
+            subsequent ``analysis_instances`` e.g. unwrapping of molecules.
+        """
+
+        # Ensure compatibility with API of version 0.15.0
+        if not hasattr(self, "_analysis_instances"):
+            self._analysis_instances = (self,)
```

Comment:
this is confusing, instead this is making a single analysis class work as a subclass of a `Collection` right?

Reply:
No, this is just a leftover from the code before. We can also remove it I think, if we do not support 0.15.0 anymore.